backend/dal/Repositories/ContactRepository.cs

```diff
@@ -0,0 +1,206 @@
+        /// <returns></returns>
+        public int Count()
+        {
+            return this.Context.PimsContactMgrVws.Count();
+        }
+
+        /// <summary>
+        /// Get an array of contacts within the specified filters.
+        /// Note that the 'contactFilter' will control the 'page' and 'quantity'.
+        /// </summary>
+        /// <param name="filter"></param>
+        /// <returns></returns>
+        public IEnumerable<PimsContactMgrVw> Get(ContactFilter filter)
+        {
+            this.User.ThrowIfNotAuthorized(Permissions.ContactView);
+            filter.ThrowIfNull(nameof(filter));
+            if (!filter.IsValid())
+            {
+                throw new ArgumentException("Argument must have a valid filter", nameof(filter));
+            }
+
+            IEnumerable<PimsContactMgrVw> contacts = GetFilteredContacts(filter);
+
+            return contacts;
+        }
+
+        /// <summary>
+        /// Get a contact with the given id.
+        /// </summary>
+        /// <param name="id">The id of the contact to retrieve</param>
+        /// <returns></returns>
+        public PimsContactMgrVw Get(string id)
+        {
+            this.User.ThrowIfNotAuthorized(Permissions.ContactView);
+            var contact = Context.PimsContactMgrVws.Where(x => x.Id == id).FirstOrDefault();
+
+            return contact;
+        }
+
+        /// <summary>
+        /// Get a page with an array of contacts within the specified filters.
+        /// Note that the 'contactFilter' will control the 'page' and 'quantity'.
+        /// </summary>
+        /// <param name="filter"></param>
+        /// <returns></returns>
+        public Paged<PimsContactMgrVw> GetPage(ContactFilter filter)
+        {
+            this.User.ThrowIfNotAuthorized(Permissions.ContactView);
+            filter.ThrowIfNull(nameof(filter));
+            if (!filter.IsValid())
+            {
+                throw new ArgumentException("Argument must have a valid filter", nameof(filter));
+            }
+            IEnumerable<PimsContactMgrVw> results = GetFilteredContacts(filter);
+
+            return new Paged<PimsContactMgrVw>(results, filter.Page, filter.Quantity, results.Count());
+        }
+
+        /// <summary>
+        /// Find the entity for the specified 'keyValues'.
+        /// </summary>
+        /// <param name="keyValues"></param>
+        /// <returns></returns>
+        public PimsContactMgrVw Find(params object[] keyValues)
+        {
+            return this.Context.Find<PimsContactMgrVw>(keyValues);
+        }
+
+        /// <summary>
+        /// Generate an SQL statement for the specified 'filter'.
+        /// </summary>
+        /// <param name="filter"></param>
+        /// <returns></returns>
+        private IEnumerable<PimsContactMgrVw> GetFilteredContacts(ContactFilter filter)
+        {
+            filter.ThrowIfNull(nameof(filter));
+
+            var query = from c in this.Context.Set<PimsContactMgrVw>()
```

Comment:
note that this logic combined with skip/take has adverse effects on the total number of results returned. I have addressed that in my upcoming changes (tests/fixes pr).